docs/guides/es2k/es2k-guide.md

```diff
@@ -0,0 +1,212 @@
+
+```bash
+source ./scripts/es2k/setup_env.sh $IPDK_RECIPE $SDE_INSTALL $DEPEND_INSTALL
+./scripts/es2k/copy_config_files.sh $IPDK_RECIPE $SDE_INSTALL
+alias sudo='sudo PATH="$PATH" HOME="$HOME" LD_LIBRARY_PATH="$LD_LIBRARY_PATH" SDE_INSTALL="$SDE_INSTALL"'
+```
+
+#### Create IPDF netdevs
+
+After installing ATE Kernel on HOST machine, install the following drivers
+to bind the network devices (netdevs) to the E2100 target.
+
+```bash
+modprobe auxiliary 
+modprobe idpf
+```
+
+#### Create P4 artifacts
+
+- P4 programs are available in the P4 SDE repository.
+
+- Obtain a copy of the P4 Compiler for ES2K and install it on your
+  development system.
+
+- Set the environment variable OUTPUT_DIR to the location where artifacts
+  should be generated and where p4 files are available
+
+```bash
+export OUTPUT_DIR=/root/p4-driver/mev_reference_p4_files/simple_l3_l4_pna
+```
+
+- Generate the artifacts using the p4c compiler installed in the previous step:
+
+```bash
+p4c-pna-xxp -I/usr/lib -I/usr/share/p4c/p4include -I/usr/share/p4c/idpf-lib \
+            $OUTPUT_DIR/simple_l3_l4_pna.p4 -o $OUTPUT_DIR/simple_l3_l4_pna.s \
+            --p4runtime-files $OUTPUT_DIR/simple_l3_l4_pna.p4info.txt \
+            --context $OUTPUT_DIR/simple_l3_l4_pna.context.json \
+            --bfrt $OUTPUT_DIR/simple_l3_l4_pna.bfrt.json
+```
+
+*Note*: The above commands will generate three files
+(`simple_l3_l4_pna.p4info.txt`, `simple_l3_l4_pna.bfrt.json`, and
+`simple_l3_l4_pna.context.json`).
+
+- Modify `simple_l3_l4_pna.conf` to specify the correct paths for
+  bfrt-config, context, and config.
+
+- Create a dummy tofino.bin file, which is needed by tdi_pipeline_builder.
+
+```bask
+touch tofino.bin
+```
+
+- TDI pipeline builder combines the artifacts generated by the p4c compiler
+  to generate a single bin file to be pushed from the controller.
+  Generate binary executable using the TDI pipeline builder command below:
+
+```bash
+./install/bin/tdi_pipeline_builder \
+    --p4c_conf_file=$OUTPUT_DIR/simple_l3_l4_pna.conf \
+    --bf_pipeline_config_binary_file=$OUTPUT_DIR/simple_l3_l4_pna.pb.bin
+```
+
+#### Set forwarding pipeline
+
+```bash
+sudo ./install/bin/p4rt-ctl set-pipe br0 $OUTPUT_DIR/simple_l3_l4_pna.pb.bin $OUTPUT_DIR/simple_l3_l4_pna.p4info.txt
+```
+
+#### Configure forwarding rule to receive traffic
+
+Add a forwarding rule to receive traffic on VSI-1 (base offset 16 + VSI ID 1) \
+when the key matches.
+
+```bash
+sudo  ./install/bin/p4rt-ctl add-entry br0 MainControlImpl.l3_l4_match_rx \
+    "hdrs.ipv4[vmeta.common.depth].protocol=0x11,vmeta.common.port_id=0,istd.direction=0,
+    hdrs.ipv4[vmeta.common.depth].src_ip="192.168.1.10",hdrs.ipv4[vmeta.common.depth].dst_ip="192.168.1.20",
+    hdrs.udp[vmeta.common.depth].sport=1000,hdrs.udp[vmeta.common.depth].dport=2000,action=MainControlImpl.send(17)"
+```
+
+ *Note*: See [p4rt-ctl Readme](../clients/p4rt-ctl.rst) for more information
```

Comment:
As of today we dont have this doc. Going forward we will have client related documents in "ipdk.recipe/docs/guides/clients/p4rt-ctl" or "ipdk.recipe/clients/p4rt-ctl" ??

Reply:
This section is not yet done, so don't review the existing doc. It will be heavily modified :)